projects/cardiovascular-patient-app/src/app/last-report-bp/last-report-bp.component.ts

```diff
@@ -0,0 +1,43 @@
+import { Component } from '@angular/core';
+import { DataLayerManagerService, formatDate, formatTime } from 'ngx-charts-on-fhir';
+import { map } from 'rxjs';
+
+interface LastReportedBPdata {
+  systolic: { date: string; value: number };
+  diastolic: { date: string; value: number };
+}
+
+@Component({
+  selector: 'last-report-bp',
+  templateUrl: './last-report-bp.component.html',
+  styleUrls: ['./last-report-bp.component.css'],
+})
+export class LastReportBPComponent{
+  lastReportedBPdata?: LastReportedBPdata;
+  isLastPriorBp: boolean = true;
+
+  constructor( private layerManager: DataLayerManagerService) { }
+
+  ngOnInit(): void {
+    this.layerManager.allLayers$
+      .pipe(
+        map((layers) =>
+          layers
+            .filter((layer) => layer.name === 'Blood Pressure')
+            .map((layer) => layer.datasets.map((data) => data.data))
+            .map((layer) => layer.map((data) => data.slice(-1)))
+        )
+      )
```

Comment:
Also, you should test this with a patient who has some home measurements because there will be more than 2 datasets. You'll need to group the datasets to combine home/office measurements. Look at HomeMeasurementSummaryService for example.

Reply:
Yes @semicolin